### DISCUSSION TITLE
Using GPU and CPU simultaneously for parameters

### DISCUSSION THREAD

**Discussion by ssharpe42** [Lightning-AI/pytorch-lightning#9363] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule]
How do you move parameters to the CPU while training on the GPU without any example tensors for `.type_as`?

Would you just make a global cpu tensor to refer to in this case?

**Comment by rohitgr7** [comment]
wouldn't just `.to('cpu')` work in this case?

**Reply by ssharpe42** [reply]
Ok thought `.to()` was just discouraged in PL

**Comment by rohitgr7** [comment]
it is but only when you create new tensors since during distributed training calling `new_tensor.cuda()` won't be correct all the time. Although you can still use `.to` calls and access CPU using `cpu` and device using `self.device`.
